service/src/main/java/skills/skillLoading/SkillsLoader.groovy

```diff
@@ -1226,11 +1238,18 @@ class SkillsLoader {
                 Date achievedOn = achievedLevelRepository.getAchievedDateByUserIdAndProjectIdAndSkillId(userId, skillDef.projectId, skillDef.skillId)
                 Date expirationDate
                 Date mostRecentlyPerformedOn
+                Date lastExpirationDate
                 int daysOfInactivityBeforeExp = 0
                 Boolean isMotivationalSkill = false
                 if (skillDefAndUserPoints.attributes && skillDefAndUserPoints.attributes.type == SkillAttributesDef.SkillAttributesType.AchievementExpiration) {
                     ExpirationAttrs expirationAttrs = skillAttributeService.convertAttrs(skillDefAndUserPoints.attributes, ExpirationAttrs)
                     expirationDate = expirationAttrs.nextExpirationDate
+                    if(!achievedOn) {
```

Comment:
should happen in a batch rather than for each skill; (load data in `subjectDataLoader.loadData` similar to `handle*` methods on lines `skills/skillLoading/SubjectDataLoader.groovy:138-142`;  would need to join to only return dates for the non-achieved skills and then associate data by skill (similiar to other handle methods)